go/database/vt/memory/state.go

```diff
@@ -85,23 +86,34 @@ func (s *State) HasEmptyStorage(addr common.Address) (bool, error) {
 }
 
 func (s *State) Apply(block uint64, update common.Update) error {
+	
+	maybeEmptyCodeAccounts := make(map[common.Address]struct{})
```

Comment:
For the big picture, I think the responsibility on what to initialize and what not should be the job of the block processor and/or the transaction context, not the state. If a new account is created, the update should contain a corresponding code update setting the account's code to the empty code.

The update contains a [CreatedAccount](https://github.com/0xsoniclabs/carmen/blob/1af2f6427380d31c0e0fe136cdee756a5bbf90db/go/common/update.go#L44) field. Would it be sufficient to set the code for these accounts to the empty code instead of tracking the "maybeEmptyCodeAccounts" here?